src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.MultiAgent.cs

```diff
@@ -833,6 +839,39 @@ private static ulong CurlSendCallback(IntPtr buffer, ulong size, ulong nitems, I
                 return Interop.Http.CURL_READFUNC_ABORT;
             }
 
+            private static void SetLowSpeedLimit(EasyRequest easy, TimeSpan timeout)
+            {
+                // If there's no send/receive timeout on the handler, the defaults will
+                // remain disabled, and we don't need to worry about setting or resetting.
+                if (!easy._handler._hasSendReceiveTimeout)
+                {
+                    return;
+                }
+
+                // We need to ensure the timeout is set appropriately, which may include
+                // disabling a timeout currently set if one was.
+
+                long limit = 0, time = 0; // 0 == disable
+
+                if (timeout != Timeout.InfiniteTimeSpan)
+                {
+                    double secs = timeout.TotalSeconds;
+                    if (secs >= 1)
+                    {
+                        limit = 1;
```

Comment:
> I probably need to set the timeout after all request content data has been sent

Okay, that matches my thinking, glad I asked the question.

> Yes. Isn't that basically the same as saying "if we don't receive a byte for 30 seconds, abort"? That's what the timeout is meant to be: if we go N seconds without getting the data requested, abort.

If it's just meant to be the gap, that's the best way that I can think of to do it using this approach.  Otherwise a Timer or using Wait with a timeout.

A malicious (aka just trying to be annoying) server could send the first one-byte packet after 25 seconds, then send one additional byte every half second or so, and draw out the read of the first header to ~8200 seconds, which isn't what I (personally) would expect from a 30 second timeout.

Conversely, with a really low timeout, like 2 seconds, you could get the first packet of a header right away, then a horrible GC happens on the server that stalls it for 3 seconds, and then it sends the next packet; but the read aborted, even though the timeout was only supposed to cover the gap.

> It's meant to be the timeout for each underlying send/receive.

Yeah, that's another place that feels weird.  Since curl is firing the callback after each successfully read header, do we really want 30 seconds per header?  If it's just the gap until "the headers" then maybe between each header we'd use the body timeout? (Since we don't know if it was the last header or not)  Or use a decaying timer so there's BodyTimeout from "start of first header" to "start of body".